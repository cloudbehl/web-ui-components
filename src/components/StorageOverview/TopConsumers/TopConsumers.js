import React from 'react';
import { LineChart, Row, Col } from 'patternfly-react';
import PropTypes from 'prop-types';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
} from '../../Dashboard/DashboardCard';
import { StorageOverviewContext } from '../StorageOverviewContext/StorageOverviewContext';

import { getTopConsumerVectorStats } from '../../../selectors';

const TopConsumersBody = ({ columns }) => (
  <div className="kubevirt-top-consumers__body">
    <Row>
      <Col lg={12} md={12} sm={12} xs={12}>
        <LineChart
          id="line-chart"
          data={{
            x: 'x',
            columns,
            type: 'line',
          }}
          axis={{
            y: {
              label: {
                text: 'Used Capacity',
                position: 'outer-top',
              },
            },
            x: {
              tick: {
                count: 9,
                format: v => `${Math.floor(v / 100)} : ${Math.floor(v % 100)}`,
              },
            },
          }}
        />
      </Col>
    </Row>
  </div>
);

export const TopConsumers = ({ stats, loaded }) => {
  let columns;

  if (loaded) columns = getTopConsumerVectorStats(stats);

  return (
    <DashboardCard>
      <DashboardCardHeader>
        <DashboardCardTitle>Top Consumers</DashboardCardTitle>
      </DashboardCardHeader>
      <DashboardCardBody isLoading={!loaded}>
        <TopConsumersBody columns={columns} />
      </DashboardCardBody>
    </DashboardCard>
  );
};

TopConsumersBody.propTypes = {
  columns: PropTypes.array.isRequired,
};

TopConsumers.defaultProps = {
  loaded: false,
};

TopConsumers.propTypes = {
  stats: PropTypes.array.isRequired,
  loaded: PropTypes.bool,
};

const TopConsumersConnected = () => (
  <StorageOverviewContext.Consumer>{props => <TopConsumers {...props} />}</StorageOverviewContext.Consumer>
);

export default TopConsumersConnected;
