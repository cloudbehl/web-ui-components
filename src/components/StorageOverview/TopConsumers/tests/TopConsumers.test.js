import React from 'react';
import { render } from 'enzyme';

import { TopConsumers } from '../TopConsumers';
import { TopConsumerStats } from '../fixtures/TopConsumers.fixture';

const testTopConsumersOverview = () => <TopConsumers {...TopConsumerStats} />;

describe('<TopConsumers />', () => {
  it('renders correctly', () => {
    const component = render(testTopConsumersOverview());
    expect(component).toMatchSnapshot();
  });
});
