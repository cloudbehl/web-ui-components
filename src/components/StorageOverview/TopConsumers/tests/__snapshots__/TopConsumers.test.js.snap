// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopConsumers /> renders correctly 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title"
    >
      Top Consumers
    </h2>
  </div>
  <div
    class="card-pf-body"
  >
    <div
      class="kubevirt-top-consumers__body"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12 col-md-12 col-sm-12 col-xs-12"
        >
          <div
            class=" line-chart-pf"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
